driver.go

```diff
@@ -295,21 +297,23 @@ func mount(accountName, accountKey, mountpoint string, options VolumeOptions) er
 		options.GID = "0"
 	}
 	mount := fmt.Sprintf("//%s.file.core.windows.net/%s", accountName, options.Share)
-	opts := fmt.Sprintf(
-		"vers=3.0,username=%s,password=%s,dir_mode=%s,file_mode=%s,uid=%s,gid=%s",
-		accountName,
-		accountKey,
-		options.DirMode,
-		options.FileMode,
-		options.UID,
-		options.GID,
-	)
+	opts = append(opts, "vers=3.0")
```

Comment:
Sorry for the delay on this, I've been super busy.  

We found that we needed to add in the boolean nolock option for some things. For mount, boolean options either exist or are left off.  Adding the options to a slice if they are true and joining that into a string seems to be a good way to do it.   

I can format the "required" options like your above comment, but any optional or boolean ones I'd still do as an append to the slice.

As a side note, I've tested this and seems to be working reliably so far.